hazelcast/src/main/java/com/hazelcast/cache/impl/AbstractCacheProxyBase.java

```diff
@@ -260,7 +262,33 @@ public void run() {
             return ownerKeys;
         }
 
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        AbstractCacheProxyBase that = (AbstractCacheProxyBase) o;
+        if (nameWithPrefix != null ? !nameWithPrefix.equals(that.nameWithPrefix) : that.nameWithPrefix != null) {
+            return false;
+        }
+
+        return true;
+    }
+
+    @Override
+    public int hashCode() {
+        return nameWithPrefix != null ? nameWithPrefix.hashCode() : 0;
+    }
 
+    @Override
+    public String toString() {
+        return getClass().getName() + '{' + "name=" + name + ", fullName=" + nameWithPrefix + '}';
```

Comment:
Is it possible to use one definition all over the code (methods and variable definitions)? For ex. using one of  `fullName` or `nameWithPrefix`